python/cudf/cudf/tests/test_string.py

```diff
@@ -361,29 +361,45 @@ def test_string_len(ps_gs):
         ["f", "g", "h", "i", "j"],
         ("f", "g", "h", "i", "j"),
         pd.Series(["f", "g", "h", "i", "j"]),
-        pd.Index(["f", "g", "h", "i", "j"]),
-        (["f", "g", "h", "i", "j"], ["f", "g", "h", "i", "j"]),
-        [["f", "g", "h", "i", "j"], ["f", "g", "h", "i", "j"]],
-        (
-            pd.Series(["f", "g", "h", "i", "j"]),
-            ["f", "a", "b", "f", "a"],
-            pd.Series(["f", "g", "h", "i", "j"]),
-            ["f", "a", "b", "f", "a"],
-            ["f", "a", "b", "f", "a"],
-            pd.Index(["1", "2", "3", "4", "5"]),
-            ["f", "a", "b", "f", "a"],
+        # xref pandas-dev/#33436
```

Comment:
There's a few issues with this test. First is an actual bug that I reported to pandas-dev, and they picked up and fixed - it should be in the next release, not sure when that is.

The second and more important issue is that in Pandas, `str.cat` performs index alignment, and I don't think we do that here. Most of the comparison data ends up NaN. I think the optimal way around this is to just move the goalposts so that when we're concatenating against a Series in the tests, that series has the same index as the source series.

Reply:
Can you raise an issue about us not doing index alignment on `str.cat` and then link to the issue in an xfail or warning message? cc @galipremsagar for visibility

Reply:
Looked into the issue, will handle it on the python side.
https://github.com/rapidsai/cudf/issues/5321

Reply:
Do we still need to link to the issue raised by Prem somewhere in this test?